openpype/pipeline/stagingdir.py

```diff
@@ -0,0 +1,162 @@
+    # backward compatibility
+    if not custom_staging_dir_profiles and not transient_data_profiles:
+        return
+
+    # backward compatibility
+    if custom_staging_dir_profiles:
+        transient_data_profiles += custom_staging_dir_profiles
+
+    if not log:
+        log = Logger.get_logger("get_transient_data_profile")
+
+    filtering_criteria = {
+        "hosts": host_name,
+        "families": family,
+        "task_names": task_name,
+        "task_types": task_type,
+        "subsets": subset_name
+    }
+    profile = filter_profiles(
+        transient_data_profiles, filtering_criteria, logger=log)
+
+    if not profile or not profile["active"]:
+        return
+
+    if not anatomy:
+        anatomy = pipeline.Anatomy(project_name)
+
+    template_name = profile["template_name"] or TRANSIENT_DIR_TEMPLATE
+    _validate_transient_template(project_name, template_name, anatomy)
+
+    transient_template = anatomy.templates[template_name]["folder"]
+    data_persistence = (
+        # TODO: make this compulsory in the future
+        profile.get("data_persistence")
+        # maintain backwards compatibility
+        or profile.get("custom_staging_dir_persistent")
+    )
+
+    return {
+        "transient_template": transient_template,
+        "transient_persistence": data_persistence
+    }
+
+
+def _validate_transient_template(project_name, template_name, anatomy):
+    """Check that transient template is correctly configured.
+
+    Raises:
+        ValueError - if misconfigured template
+    """
+    if template_name not in anatomy.templates:
+        raise ValueError((
+            "Anatomy of project \"{}\" does not have set"
+            " \"{}\" template key!").format(project_name, template_name)
+        )
+
+    if "folder" not in anatomy.templates[template_name]:
+        raise ValueError((
+            "There is not set \"folder\" template in \"{}\" anatomy"
+            " for project \"{}\".").format(template_name, project_name)
+        )
+
+
+def get_instance_staging_dir(instance):
+    """Unified way how staging dir is stored and created on instances.
+
+    First check if 'stagingDir' is already set in instance data.
+    In case there already is new tempdir will not be created.
+
+    It also supports `OPENPYPE_TMPDIR`, so studio can define own temp
+    shared repository per project or even per more granular context.
+    Template formatting is supported also with optional keys. Folder is
+    created in case it doesn't exists.
+
+    Available anatomy formatting keys:
+        - root[work | <root name key>]
+        - project[name | code]
+
+    Note:
+        Staging dir does not have to be necessarily in tempdir so be careful
+        about its usage.
+
+    Args:
+        instance (pyblish.lib.Instance): Instance for which we want to get
+            staging dir.
+
+    Returns:
+        str: Path to staging dir of instance.
+    """
+    staging_dir = instance.data.get('stagingDir')
+    if staging_dir:
```

Comment:
Would it be worth checking if the `stagingDir` actually exists on disk?

Reply:
At this stage path should exists. But yeah since I need it to prove it, I have found out that `CollectManagedStagingDir` plugin was missing the `os.makedirs`. Thanks @Minkiu !